src/subtitles/doc.md

```diff
@@ -0,0 +1,206 @@
+
+The system identifies subtitle tracks through multiple methods:
+
+1. **Handler Type Check**:
+   ```rust
+   // In hdlr box (bytes 8-12)
+   let handler_type = std::str::from_utf8(&hdlr_data[8..12]);
+   // "sbtl", "subt", "text" indicate subtitle tracks
+   ```
+
+2. **Media Info Check**:
+   ```rust
+   // In minf box
+   find_box(minf_data, "sbtl")  // Subtitle handler
+   find_box(minf_data, "subt")  // Subtitle handler
+   find_box(minf_data, "text")  // Text handler
+   ```
+
+## Extraction Process
+
+### 1. Format Detection
+- Skip MP3 files (no subtitles)
+- Support MP4 family formats (MP4, M4V, 3GP, 3G2, MOV)
+- Proceed with extraction for compatible formats
+
+### 2. Track Analysis
+- Locate `moov` box efficiently (8KB start/end search)
+- Navigate `moov → trak → mdia → minf → stbl` hierarchy
+- Identify subtitle tracks via `hdlr` box (handler type "sbtl"/"subt"/"text")
+
+### 3. Sample Table Processing
+- **STTS**: Sample timing and duration
+- **STSZ**: Sample sizes for byte range calculation
+- **STSC**: Sample-to-chunk mapping
+- **STCO/CO64**: Chunk offset locations
+
+### 4. Optimized Download
+- Calculate precise byte ranges for each subtitle sample
+- Group nearby ranges (gap < 4KB) to minimize HTTP requests
+- Download optimized chunks and parse subtitle data
+
+### 5. Format Parsing
+- Route to format-specific parsers based on codec type
+- Extract text content and apply default 2-second duration
+- Format timestamps in SRT standard (HH:MM:SS,mmm)
+
+## Supported Formats
+
+### TX3G (3GPP Timed Text)
+- **Structure**: 2-byte text length + UTF-8 text data
+- **Example**: `[00 1E] [text bytes...]` (30 bytes of text)
+
+### WebVTT
+- **Structure**: UTF-8 text with time markers
+- **Parsing**: Skip "WEBVTT" header, extract content
+
+### TTML (Timed Text Markup Language)
+- **Structure**: XML-based format
+- **Parsing**: Strip XML tags, extract text content
+
+### Generic
+- **Fallback**: UTF-8 first, then UTF-16
+- **Use**: Unknown or unsupported codec types
+
+## Usage Examples
+
+### Local File Extraction
+```rust
+use mediaparser::extract_subtitles;
+
+let entries = extract_subtitles("video.mp4").await;
+for entry in entries {
+    println!("{} --> {}", entry.start, entry.end);
+    println!("{}", entry.text);
+}
+```
+
+### Remote File Extraction
+```rust
+use mediaparser::extract_subtitles;
+
+let entries = extract_subtitles("https://example.com/video.mp4").await;
+println!("Extracted {} subtitle entries", entries.len());
+```
+
+### Direct Sample Parsing
+```rust
+use mediaparser::subtitles::{parse_subtitle_sample_data, format_timestamp};
+
+let sample_data = b"\x00\x0AHello World";
+let entries = parse_subtitle_sample_data(sample_data, 5.0, "tx3g").await;
+// Returns: SubtitleEntry { start: "00:00:05,000", end: "00:00:07,000", text: "Hello World" }
+```
+
+## Output Format
+
+### SubtitleEntry Structure
+```rust
+pub struct SubtitleEntry {
+    pub start: String,    // SRT format: "00:00:05,000"
```

Comment:
Are the timestamp formats going to vary across subtitle formats? If so, should we either:

1. Normalize to a common format
2. Include the subtitle type in `extract_subtitles`'s return value

Probably it would nice to include #2 in any case